subprojects/common/graph/src/main/kotlin/com/avito/graph/Operation.kt

```diff
@@ -0,0 +1,14 @@
+package com.avito.graph
+
+interface Operation {
+    val id: String
+    val duration: Double
+
+    /**
+     * ids of predecessor operations
+     */
+    val predecessors: Set<String>
```

Comment:
I did not meet a critical path vocabulary, is it a common name for graph relations?

Reply:
In this case, "predecessor" defines incoming edges.
I found usage of this term in similar cases: in Gradle build scan and in the Activity-on-node diagram.

- https://en.wikipedia.org/wiki/Critical_path_method#Visualizing_critical_path_schedule
- https://project-management-knowledge.com/definitions/a/activity-on-node/